Tests/Functional/Overall/DataSyncTest.php

```diff
@@ -0,0 +1,213 @@
+<?php
+
+/*
+ * This file is part of the ONGR package.
+ *
+ * (c) NFQ Technologies UAB <info@nfq.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ONGR\OXIDConnectorBundle\Tests\Functional\Overall;
+
+use ONGR\ConnectionsBundle\Command\SyncExecuteCommand;
+use ONGR\ConnectionsBundle\Command\SyncProvideCommand;
+use ONGR\OXIDConnectorBundle\Tests\Functional\TestBase;
+use ONGR\ConnectionsBundle\Sync\StorageManager\MysqlStorageManager;
+use ONGR\ConnectionsBundle\Service\PairStorage;
+use ONGR\ConnectionsBundle\Sync\DiffProvider\Binlog\BinlogDiffProvider;
+use ONGR\ConnectionsBundle\Sync\DiffProvider\Binlog\BinlogParser;
+use ONGR\ElasticsearchBundle\DSL\Search;
+use ONGR\ElasticsearchBundle\ORM\Repository;
+
+class DataSyncTest extends TestBase{
+
+    /**
+     * Tests data importing from client data to sync storage.
+     */
+    public function testDataImportFromClientDataToSyncStorage()
+    {
+        // Delete existing binlog data.
+        $this->getConnection()->executeQuery('RESET MASTER');
+
+        $this->importData('DataImportTest/updateProducts.sql');
+        $this->importData('DataImportTest/updateCategories.sql');
+        $this->importData('DataImportTest/updateContent.sql');
+
+        /** @var MysqlStorageManager $managerMysql */
+        $managerMysql = $this
+            ->getServiceContainer()
+            ->get('ongr_connections.sync.storage_manager.mysql_storage_manager');
+
+        // Create storage for shop.
+        $managerMysql->createStorage(0);
+
+        // Set initial start position, if not sure, 0 always returns results.
+        $this->setLastSync(0, BinlogParser::START_TYPE_POSITION);
+
+        // Set service so, that it would use last sync position.
+        $this
+            ->getServiceContainer()
+            ->get('ongr_connections.sync.diff_provider.binlog_diff_provider')
+            ->setStartType(BinlogParser::START_TYPE_POSITION);
+
+        // Then start data import with pipeline $target.
```

Comment:
"data provide" is name of process, I think "data import" discribes this process better.